yarn-project/end-to-end/src/e2e_public_to_private_messaging.test.ts

```diff
@@ -38,49 +41,59 @@ describe('e2e_public_to_private_messaging', () => {
 
   const deployContract = async () => {
     logger(`Deploying Public to Private L2 contract...`);
-    const deployer = new ContractDeployer(PublicToPrivateContractAbi, aztecRpcServer);
-    const tx = deployer.deploy().send();
+    const deployer = new ContractDeployer(NonNativeTokenContractAbi, aztecRpcServer);
+    const tx = deployer.deploy(0n, ownerPub).send();
     const receipt = await tx.getReceipt();
-    contract = new Contract(receipt.contractAddress!, PublicToPrivateContractAbi, aztecRpcServer);
+    contract = new Contract(receipt.contractAddress!, NonNativeTokenContractAbi, aztecRpcServer);
     await tx.isMined(0, 0.1);
     await tx.getReceipt();
     logger('L2 contract deployed');
     return contract;
   };
 
   /**
-   * Milestone 5.4: Intra-contract Public -\> Private calls.
+   * Milestone 5.4: Intra-contract Public -\> Private calls (shielding).
    */
-  it('5.4: Should be able to create a commitment in a public function and spend in a private function', async () => {
+  it('5.4: Should be able to create a private commitment in a public function, then convert back to public', async () => {
     const mintAmount = 100n;
 
-    const [owner, receiver] = accounts;
+    const [owner] = accounts;
 
-    const deployedContract = await deployContract();
+    await deployContract();
```

Comment:
Before shielding, can you add a check to see that the supply matches the expected. Also should probably mint something, it seems strange that you can shield something, when there is nothing to shield.